packages/ssr-plugin/example/public/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { h, render } from 'preact';
+import { LocationProvider, Router, ErrorBoundary, lazy, hydrate } from 'preact-iso';
+import Home from './pages/home.js';
+import NotFound from './pages/_404.js';
+import Header from './header.tsx';
+
+const About = lazy(() => import('./pages/about/index.js'));
+const LazyAndLate = lazy(
+	() =>
+		new Promise(r => {
+			setTimeout(() => {
+				r(import('./pages/about/index.js'));
+			}, 1.5e3);
+		})
+);
+const CompatPage = lazy(() => import('./pages/compat.js'));
+const ClassFields = lazy(() => import('./pages/class-fields.js'));
+const Files = lazy(() => import('./pages/files/index.js'));
+const Environment = lazy(async () => (await import('./pages/environment/index.js')).Environment);
+
+export function App() {
+	return (
+		<LocationProvider>
+			<div class="app">
+				<Header />
+				<ErrorBoundary>
+					<Router>
+						<Home path="/" />
+						<About path="/about" />
+						<LazyAndLate path="/lazy-and-late" title={'Lazy and Late'} />
+						<CompatPage path="/compat" />
+						<ClassFields path="/class-fields" />
+						<Files path="/files" />
+						<Environment path="/env" />
+						<NotFound default />
+					</Router>
+				</ErrorBoundary>
+			</div>
+		</LocationProvider>
+	);
+}
+
+if (typeof window !== 'undefined') {
+	if (document.querySelector('.app')) {
+		hydrate(<App />, document.body);
+	} else {
+		render(<App />, document.body);
+	}
+}
```

Comment:
Is this necessary, given that `hydrate` is a wrapper that calls `render` if `! document.querySelector('script[type=isodata]')`? (I suppose the answer is yes if `script[type=isodata]` isn't generated by the SSR method implemented in this PR, but in this case could this be clarified in a code comment, for the avoidance of doubt?)

Code reference:

https://github.com/preactjs/wmr/blob/fe51b7eb3a65740571863fbeebf3367b74a9ddfe/packages/preact-iso/hydrate.js#L8-L15

The current documentation states:

https://github.com/preactjs/wmr/blob/main/packages/preact-iso/README.md#hydratejs

> hydrate() is a thin wrapper around Preact's hydrate() method. It performs hydration when the HTML for the current page includes pre-rendered output from prerender(). It falls back to plain rendering in any other cases, which is useful if you're not pre-rendering during development. This method also checks to make sure its running in a browser context before attempting any rendering - if not, it does nothing.